diff -up ncurses-6.3-20221126/misc/gen-pkgconfig.in.config ncurses-6.3-20221126/misc/gen-pkgconfig.in
--- ncurses-6.3-20221126/misc/gen-pkgconfig.in.config	2022-10-08 18:45:20.000000000 +0200
+++ ncurses-6.3-20221126/misc/gen-pkgconfig.in	2022-11-29 17:04:43.353766420 +0100
@@ -83,7 +83,7 @@ if [ "$includedir" != "/usr/include" ];
 fi
 
 lib_flags=
-for opt in -L$libdir @EXTRA_PKG_LDFLAGS@ @LIBS@
+for opt in -L$libdir @LIBS@
 do
 	case $opt in
 	-l*) # LIBS is handled specially below
diff -up ncurses-6.3-20221126/misc/ncurses-config.in.config ncurses-6.3-20221126/misc/ncurses-config.in
--- ncurses-6.3-20221126/misc/ncurses-config.in.config	2022-07-26 23:36:28.000000000 +0200
+++ ncurses-6.3-20221126/misc/ncurses-config.in	2022-11-29 17:06:04.381597412 +0100
@@ -41,7 +41,6 @@ exec_prefix="@exec_prefix@"
 
 bindir="@bindir@"
 includedir="@includedir@"
-libdir="@libdir@"
 datarootdir="@datarootdir@"
 datadir="@datadir@"
 mandir="@mandir@"
@@ -101,7 +100,7 @@ fi
 # There is no portable way to find the list of standard library directories.
 # Require a POSIX shell anyway, to keep this simple.
 lib_flags=
-for opt in -L$libdir @EXTRA_PKG_LDFLAGS@ $LIBS
+for opt in $LIBS
 do
 	case $opt in
 	-specs*) # ignore linker specs-files which were used to build library
@@ -120,13 +119,13 @@ do
 		lib_check=`echo "x$opt" | sed -e 's/^.-L//'`
 		[ -d "$lib_check" ] || continue
 		case "$lib_check" in
-		@LD_SEARCHPATH@) # skip standard libdir
+		////) # skip standard libdir (disabled for multilib)
 			if [ "$lib_check" = "$libdir" ]
 			then
 				lib_first=yes
 				IFS_save="$IFS"
 				IFS='|'
-				LIBDIRS="@LD_SEARCHPATH@"
+				LIBDIRS=""
 				for lib_check in $LIBDIRS
 				do
 					if [ -d "$lib_check" ]
@@ -274,7 +273,6 @@ ENDECHO
 		echo "$INCS"
 		;;
 	--libdir)
-		echo "${libdir}"
 		;;
 	--mandir)
 		echo "${mandir}"
