diff -up ncurses-6.2-20210306/c++/Makefile.in.libs ncurses-6.2-20210306/c++/Makefile.in
--- ncurses-6.2-20210306/c++/Makefile.in.libs	2021-01-23 21:42:08.000000000 +0100
+++ ncurses-6.2-20210306/c++/Makefile.in	2021-03-11 12:02:29.576741101 +0100
@@ -118,7 +118,7 @@ SHLIB_LIST	= $(SHLIB_DIRS) \
 		-l@FORM_NAME@@USE_LIB_SUFFIX@ \
 		-l@MENU_NAME@@USE_LIB_SUFFIX@ \
 		-l@PANEL_NAME@@USE_LIB_SUFFIX@ \
-		-lncurses@USE_LIB_SUFFIX@ @SHLIB_LIST@
+		-lncurses@USE_LIB_SUFFIX@ #@SHLIB_LIST@
 
 LIBROOT		= ncurses++
 
@@ -153,8 +153,7 @@ LDFLAGS_SHARED	= $(TEST_LDFLAGS) $(CFLAG
 LDFLAGS_DEFAULT	= $(LINK_@DFT_UPR_MODEL@) $(LDFLAGS_@DFT_UPR_MODEL@)
 
 # flags for library built by this makefile
-LDFLAGS		= $(TEST_ARGS) @LDFLAGS@ \
-	@LD_MODEL@ $(TEST_LIBS) @LIBS@ $(CXXLIBS)
+LDFLAGS		= @LDFLAGS@ @LD_MODEL@ @LIBS@ $(CXXLIBS)
 
 AUTO_SRC	= \
 		etip.h
diff -up ncurses-6.2-20210306/form/Makefile.in.libs ncurses-6.2-20210306/form/Makefile.in
--- ncurses-6.2-20210306/form/Makefile.in.libs	2021-01-23 21:42:08.000000000 +0100
+++ ncurses-6.2-20210306/form/Makefile.in	2021-03-11 12:00:59.001470707 +0100
@@ -110,7 +110,7 @@ LINK		= $(LIBTOOL_LINK)
 LDFLAGS		= @LDFLAGS@ @LD_MODEL@ @LIBS@
 
 SHLIB_DIRS	= -L../lib
-SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@USE_LIB_SUFFIX@ @SHLIB_LIST@
+SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@USE_LIB_SUFFIX@ #@SHLIB_LIST@
 
 RPATH_LIST	= @RPATH_LIST@
 RESULTING_SYMS	= @RESULTING_SYMS@
diff -up ncurses-6.2-20210306/menu/Makefile.in.libs ncurses-6.2-20210306/menu/Makefile.in
--- ncurses-6.2-20210306/menu/Makefile.in.libs	2020-08-29 16:50:45.000000000 +0200
+++ ncurses-6.2-20210306/menu/Makefile.in	2021-03-11 12:00:59.002470754 +0100
@@ -110,7 +110,7 @@ LINK		= $(LIBTOOL_LINK)
 LDFLAGS		= @LDFLAGS@ @LD_MODEL@ @LIBS@
 
 SHLIB_DIRS	= -L../lib
-SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@USE_LIB_SUFFIX@ @SHLIB_LIST@
+SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@USE_LIB_SUFFIX@ #@SHLIB_LIST@
 
 RPATH_LIST	= @RPATH_LIST@
 RESULTING_SYMS	= @RESULTING_SYMS@
diff -up ncurses-6.2-20210306/panel/Makefile.in.libs ncurses-6.2-20210306/panel/Makefile.in
--- ncurses-6.2-20210306/panel/Makefile.in.libs	2020-08-29 16:50:45.000000000 +0200
+++ ncurses-6.2-20210306/panel/Makefile.in	2021-03-11 12:00:59.002470754 +0100
@@ -112,7 +112,7 @@ LINK		= $(LIBTOOL_LINK)
 LDFLAGS		= @LDFLAGS@ @LD_MODEL@ @LIBS@
 
 SHLIB_DIRS	= -L../lib
-SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@USE_LIB_SUFFIX@ @SHLIB_LIST@
+SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@USE_LIB_SUFFIX@ #@SHLIB_LIST@
 
 RPATH_LIST	= @RPATH_LIST@
 RESULTING_SYMS	= @RESULTING_SYMS@
