diff -up ncurses-6.4-20230107/misc/terminfo.src.kbs ncurses-6.4-20230107/misc/terminfo.src
--- ncurses-6.4-20230107/misc/terminfo.src.kbs	2023-01-09 14:47:05.097093771 +0100
+++ ncurses-6.4-20230107/misc/terminfo.src	2023-01-09 14:47:05.100093766 +0100
@@ -6711,7 +6711,7 @@ rxvt-basic|rxvt terminal base (X Window
 	enacs=\E(B\E)0, flash=\E[?5h$<100/>\E[?5l, home=\E[H,
 	ht=^I, hts=\EH, ich=\E[%p1%d@, il=\E[%p1%dL, il1=\E[L,
 	ind=\n, is1=\E[?47l\E=\E[?1l,
-	is2=\E[r\E[m\E[2J\E[H\E[?7h\E[?1;3;4;6l\E[4l, kbs=^H,
+	is2=\E[r\E[m\E[2J\E[H\E[?7h\E[?1;3;4;6l\E[4l,
 	kcbt=\E[Z, kmous=\E[M, rc=\E8, rev=\E[7m, ri=\EM, rmacs=^O,
 	rmir=\E[4l, rmkx=\E>, rmso=\E[27m, rmul=\E[24m,
 	rs1=\E>\E[1;3;4;5;6l\E[?7h\E[m\E[r\E[2J\E[H,
@@ -6722,7 +6722,7 @@ rxvt-basic|rxvt terminal base (X Window
 	    %p9%t\016%e\017%;,
 	sgr0=\E[0m\017, smacs=^N, smir=\E[4h, smkx=\E=, smso=\E[7m,
 	smul=\E[4m, tbc=\E[3g, use=xterm+alt47, use=vt100+enq,
-	use=rxvt+pcfkeys, use=vt220+cvis, use=vt220+keypad,
+	use=rxvt+pcfkeys, use=vt220+cvis, use=vt220+keypad, use=xterm+kbs,
 # Key Codes from rxvt reference:
 #
 # Note: Shift + F1-F10 generates F11-F20
@@ -8523,6 +8523,7 @@ screen.xterm-r6|screen customized for X1
 # on Solaris because Sun's curses implementation gets confused.
 screen.teraterm|disable ncv in teraterm,
 	ncv#127,
+	kbs=^H,
 	acsc=+\020\,\021-\030.^Y0\333`\004a\261f\370g\361h\260i
 	     \316j\331k\277l\332m\300n\305o~p\304q\304r\304s_t\303u
 	     \264v\301w\302x\263y\363z\362{\343|\330}\234~\376,
