From c4f1d3da6387b2d4b6ed6c37a71d9e2a6e65bab7 Mon Sep 17 00:00:00 2001
From: Ben Cressey <bcressey@amazon.com>
Date: Tue, 21 Sep 2021 00:03:42 +0000
Subject: [PATCH] use prefix from environment

Signed-off-by: Ben Cressey <bcressey@amazon.com>
Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 2c7cc93..54bf0a0 100644
--- a/Makefile
+++ b/Makefile
@@ -24,7 +24,7 @@ WITH_SECCOMP ?= yes
 
 ##### Global definitions #####
 
-export prefix      = /usr/local
+export prefix      ?= /usr/local
 export exec_prefix = $(prefix)
 export bindir      = $(exec_prefix)/bin
 export libdir      = $(exec_prefix)/lib
-- 
2.37.1

