From 831d6c71e30b8d533123600648bb12cbb225a100 Mon Sep 17 00:00:00 2001
From: Arnaldo Garcia Rincon <agarrcia@amazon.com>
Date: Mon, 1 Nov 2021 16:18:19 +0000
Subject: [PATCH] Use NVIDIA_PATH to look up binaries

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>
---
 src/nvc_info.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/nvc_info.c b/src/nvc_info.c
index 55c3ba9..56dcbd2 100644
--- a/src/nvc_info.c
+++ b/src/nvc_info.c
@@ -251,8 +251,8 @@ find_binary_paths(struct error *err, struct dxcore_context* dxcore, struct nvc_d
         char path[PATH_MAX];
         int rv = -1;
 
-        if ((env = secure_getenv("PATH")) == NULL) {
-                error_setx(err, "environment variable PATH not found");
+        if ((env = secure_getenv("NVIDIA_PATH")) == NULL && (env = secure_getenv("PATH")) == NULL) {
+                error_setx(err, "environment variables NVIDIA_PATH/PATH not found");
                 return (-1);
         }
         if ((env = ptr = xstrdup(err, env)) == NULL)
-- 
2.37.1

