From 89af88d823a01785947f49cfda7c9f9db541489a Mon Sep 17 00:00:00 2001
From: Ben Cressey <bcressey@amazon.com>
Date: Fri, 20 Sep 2019 00:33:47 +0000
Subject: [PATCH 2/3] do not omit debug info

Signed-off-by: Ben Cressey <bcressey@amazon.com>
---
 hack/lib/golang.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hack/lib/golang.sh b/hack/lib/golang.sh
index a82880a5..5454e73f 100755
--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -779,7 +779,7 @@ kube::golang::build_binaries() {
     host_platform=$(kube::golang::host_platform)
 
     local goflags goldflags goasmflags gogcflags
-    goldflags="${GOLDFLAGS:-} -s -w $(kube::version::ldflags)"
+    goldflags="${GOLDFLAGS:-} $(kube::version::ldflags)"
     goasmflags="-trimpath=${KUBE_ROOT}"
     gogcflags="${GOGCFLAGS:-} -trimpath=${KUBE_ROOT}"
 
-- 
2.21.0

