From 725982e848e600ed6f5028fb2197a5ffe0070806 Mon Sep 17 00:00:00 2001
From: Michael Marineau <michael.marineau@coreos.com>
Date: Mon, 22 Aug 2016 16:44:30 -0700
Subject: [PATCH] gptrepair_test: fix typo in cleanup trap

---
 tests/gptrepair_test.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/gptrepair_test.in b/tests/gptrepair_test.in
index 80b2de6..805dc17 100644
--- a/tests/gptrepair_test.in
+++ b/tests/gptrepair_test.in
@@ -53,7 +53,7 @@ case "${grub_modinfo_target_cpu}-${grub_modinfo_platform}" in
 esac
 img1="`mktemp "${TMPDIR:-/tmp}/tmp.XXXXXXXXXX"`" || exit 1
 img2="`mktemp "${TMPDIR:-/tmp}/tmp.XXXXXXXXXX"`" || exit 1
-trap "rm -f '${img1}' '${ing2}'" EXIT
+trap "rm -f '${img1}' '${img2}'" EXIT
 
 create_disk_image () {
     size=$1
-- 
2.21.3

