From d25e2c8d5cb0778ae87ad43b1f4c301abe5a932b Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Sat, 23 Dec 2023 06:24:41 -0800
Subject: [PATCH] NEWS: Mention bug fixes for 29039/30694/30709/30721

---
 NEWS | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/NEWS b/NEWS
index 6fbb8a9e1d..db4d6c8373 100644
--- a/NEWS
+++ b/NEWS
@@ -31,6 +31,10 @@ Security related changes:
 
 The following bugs are resolved with this release:
 
+  [29039] Corrupt DTV after reuse of a TLS module ID following dlclose with unused TLS
+  [30694] The iconv program no longer tells the user which given encoding name was wrong
+  [30709] nscd fails to build with cleanup handler if built with -fexceptions
+  [30721] x86_64: Fix build with --disable-multiarch
   [30723] posix_memalign repeatedly scans long bin lists
   [30789] sem_open will fail on multithreaded scenarios when semaphore
     file doesn't exist (O_CREAT)
-- 
2.43.0

