From 27339a3eb8f987eebae72b854af80256c1588ebd Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Sat, 23 Dec 2023 06:27:50 -0800
Subject: [PATCH] NEWS: Mention bug fixes for 30745/30843

---
 NEWS | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/NEWS b/NEWS
index db4d6c8373..905230b838 100644
--- a/NEWS
+++ b/NEWS
@@ -36,11 +36,13 @@ The following bugs are resolved with this release:
   [30709] nscd fails to build with cleanup handler if built with -fexceptions
   [30721] x86_64: Fix build with --disable-multiarch
   [30723] posix_memalign repeatedly scans long bin lists
+  [30745] Slight bug in cache info codes for x86
   [30789] sem_open will fail on multithreaded scenarios when semaphore
     file doesn't exist (O_CREAT)
   [30804] F_GETLK, F_SETLK, and F_SETLKW value change for powerpc64 with
     -D_FILE_OFFSET_BITS=64
   [30842] Stack read overflow in getaddrinfo in no-aaaa mode (CVE-2023-4527)
+  [30843] potential use-after-free in getcanonname (CVE-2023-4806)
   [31184] FAIL: elf/tst-tlsgap
   [31185] Incorrect thread point access in _dl_tlsdesc_undefweak and _dl_tlsdesc_dynamic
 
-- 
2.43.0

