From 321c1e77a30727c1faa1d6c1345e27d54bac15bd Mon Sep 17 00:00:00 2001
From: Samuel Karp <skarp@amazon.com>
Date: Thu, 16 Jul 2020 10:52:01 -0700
Subject: [PATCH 2/4] bottlerocket: version values settable with linker

---
 agent/version/version.go | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/agent/version/version.go b/agent/version/version.go
index a89f70da..80c7b249 100644
--- a/agent/version/version.go
+++ b/agent/version/version.go
@@ -22,10 +22,10 @@ package version
 // repository. Only the 'Version' const should change in checked-in source code
 
 // Version is the version of the Agent
-const Version = "1.47.0"
+var Version = ""
 
 // GitDirty indicates the cleanliness of the git repo when this agent was built
-const GitDirty = true
+const GitDirty = false
 
 // GitShortHash is the short hash of this agent build
-const GitShortHash = "006b065d"
+var GitShortHash = ""
-- 
2.23.3

