From 331c76247d4c4c0f8fb968e21db85e28b991b716 Mon Sep 17 00:00:00 2001
From: "Sean P. Kelly" <seankell@amazon.com>
Date: Wed, 1 Dec 2021 19:19:52 +0000
Subject: [PATCH 2/5] bottlerocket: remove unsupported capabilities

---
 agent/app/agent_capability.go | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/agent/app/agent_capability.go b/agent/app/agent_capability.go
index 5febc49..48cf468 100644
--- a/agent/app/agent_capability.go
+++ b/agent/app/agent_capability.go
@@ -233,36 +233,36 @@ func (agent *ecsAgent) capabilities() ([]*ecs.Attribute, error) {
 	capabilities = agent.appendPIDAndIPCNamespaceSharingCapabilities(capabilities)
 
 	// ecs agent version 1.26.0 supports aws-appmesh cni plugin
-	capabilities = agent.appendAppMeshCapabilities(capabilities)
+	// capabilities = agent.appendAppMeshCapabilities(capabilities)
 
 	// support elastic inference in agent
-	capabilities = agent.appendTaskEIACapabilities(capabilities)
+	// capabilities = agent.appendTaskEIACapabilities(capabilities)
 
 	// support aws router capabilities for fluentd
-	capabilities = agent.appendFirelensFluentdCapabilities(capabilities)
+	// capabilities = agent.appendFirelensFluentdCapabilities(capabilities)
 
 	// support aws router capabilities for fluentbit
-	capabilities = agent.appendFirelensFluentbitCapabilities(capabilities)
+	// capabilities = agent.appendFirelensFluentbitCapabilities(capabilities)
 
 	// support aws router capabilities for log driver router
-	capabilities = agent.appendFirelensLoggingDriverCapabilities(capabilities)
+	// capabilities = agent.appendFirelensLoggingDriverCapabilities(capabilities)
 
 	// support aws router capabilities for log driver router config
-	capabilities = agent.appendFirelensLoggingDriverConfigCapabilities(capabilities)
+	// capabilities = agent.appendFirelensLoggingDriverConfigCapabilities(capabilities)
 
 	// support efs on ecs capabilities
 	capabilities = agent.appendEFSCapabilities(capabilities)
 
 	// support external firelens config
-	capabilities = agent.appendFirelensConfigCapabilities(capabilities)
+	// capabilities = agent.appendFirelensConfigCapabilities(capabilities)
 
 	// support GMSA capabilities
 	capabilities = agent.appendGMSACapabilities(capabilities)
 
 	// support efs auth on ecs capabilities
-	for _, cap := range agent.cfg.VolumePluginCapabilities {
-		capabilities = agent.appendEFSVolumePluginCapabilities(capabilities, cap)
-	}
+	// for _, cap := range agent.cfg.VolumePluginCapabilities {
+	//	capabilities = agent.appendEFSVolumePluginCapabilities(capabilities, cap)
+	// }
 
 	// support fsxWindowsFileServer on ecs capabilities
 	capabilities = agent.appendFSxWindowsFileServerCapabilities(capabilities)
-- 
2.32.0

