From af86a52ad8de5ed133f30d9f09937ffd08264d6f Mon Sep 17 00:00:00 2001
From: Samuel Karp <skarp@amazon.com>
Date: Thu, 16 Jul 2020 10:52:01 -0700
Subject: [PATCH 2/2] bottlerocket: version values settable with linker

---
 agent/version/version.go | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/agent/version/version.go b/agent/version/version.go
index 42ea50df..80c7b249 100644
--- a/agent/version/version.go
+++ b/agent/version/version.go
@@ -22,10 +22,10 @@ package version
 // repository. Only the 'Version' const should change in checked-in source code
 
 // Version is the version of the Agent
-const Version = "1.43.0"
+var Version = ""
 
 // GitDirty indicates the cleanliness of the git repo when this agent was built
-const GitDirty = true
+const GitDirty = false
 
 // GitShortHash is the short hash of this agent build
-const GitShortHash = "76dd8ed0"
+var GitShortHash = ""
-- 
2.28.0

