From effeecb19f10b639608bf3b5a5b09d1dbd19abf9 Mon Sep 17 00:00:00 2001
From: Sean McGinnis <stmcg@amazon.com>
Date: Tue, 21 Feb 2023 19:06:43 +0000
Subject: [PATCH] Create capabilities diff

Signed-off-by: Sean McGinnis <stmcg@amazon.com>
---
 agent/app/agent_capability.go | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/agent/app/agent_capability.go b/agent/app/agent_capability.go
index 601879cd..d965f8af 100644
--- a/agent/app/agent_capability.go
+++ b/agent/app/agent_capability.go
@@ -238,10 +238,10 @@ func (agent *ecsAgent) capabilities() ([]*ecs.Attribute, error) {
 	capabilities = agent.appendPIDAndIPCNamespaceSharingCapabilities(capabilities)
 
 	// ecs agent version 1.26.0 supports aws-appmesh cni plugin
-	capabilities = agent.appendAppMeshCapabilities(capabilities)
+	// capabilities = agent.appendAppMeshCapabilities(capabilities)
 
 	// support elastic inference in agent
-	capabilities = agent.appendTaskEIACapabilities(capabilities)
+	// capabilities = agent.appendTaskEIACapabilities(capabilities)
 
 	// support aws router capabilities for fluentd
 	capabilities = agent.appendFirelensFluentdCapabilities(capabilities)
@@ -265,9 +265,9 @@ func (agent *ecsAgent) capabilities() ([]*ecs.Attribute, error) {
 	capabilities = agent.appendGMSACapabilities(capabilities)
 
 	// support efs auth on ecs capabilities
-	for _, cap := range agent.cfg.VolumePluginCapabilities {
-		capabilities = agent.appendEFSVolumePluginCapabilities(capabilities, cap)
-	}
+	// for _, cap := range agent.cfg.VolumePluginCapabilities {
+	// 	capabilities = agent.appendEFSVolumePluginCapabilities(capabilities, cap)
+	// }
 
 	// support fsxWindowsFileServer on ecs capabilities
 	capabilities = agent.appendFSxWindowsFileServerCapabilities(capabilities)
-- 
2.34.1

