From c4cc8914418857649958fabd789b6e61453a0ada Mon Sep 17 00:00:00 2001
From: "Sean P. Kelly" <seankell@amazon.com>
Date: Wed, 23 Mar 2022 11:39:32 +0000
Subject: [PATCH 8/8] bottlerocket Add missing go build tags

---
 agent/async/ttl_cache_test.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/agent/async/ttl_cache_test.go b/agent/async/ttl_cache_test.go
index 0a319d7..dbf6794 100644
--- a/agent/async/ttl_cache_test.go
+++ b/agent/async/ttl_cache_test.go
@@ -1,4 +1,5 @@
 //go:build unit
+// +build unit
 
 // Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
 //
-- 
2.32.0

