From 429c4468b0058d9c2e2fffbf6660b0f1581af6af Mon Sep 17 00:00:00 2001
From: Miroslav Lichvar <mlichvar@redhat.com>
Date: Thu, 24 Oct 2019 12:42:02 +0200
Subject: sys_linux: allow F_GETFL in seccomp filter

This is needed for fdopen().

diff --git a/sys_linux.c b/sys_linux.c
index 1f36696..63eb8f1 100644
--- a/sys_linux.c
+++ b/sys_linux.c
@@ -529,7 +529,7 @@ SYS_Linux_EnableSystemCallFilter(int level)
 #endif
   };
 
-  const static int fcntls[] = { F_GETFD, F_SETFD, F_SETFL };
+  const static int fcntls[] = { F_GETFD, F_SETFD, F_GETFL, F_SETFL };
 
   const static unsigned long ioctls[] = {
     FIONREAD, TCGETS,
-- 
cgit v0.10.2

