From d9eb89767267d43f13c30ce74f6e9c34c4000274 Mon Sep 17 00:00:00 2001
From: Ben Cressey <bcressey@amazon.com>
Date: Sat, 28 May 2022 03:52:28 +0000
Subject: [PATCH] units: mount /tmp with noexec

Signed-off-by: Ben Cressey <bcressey@amazon.com>
---
 units/tmp.mount | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/units/tmp.mount b/units/tmp.mount
index 734acea..e4990b7 100644
--- a/units/tmp.mount
+++ b/units/tmp.mount
@@ -22,4 +22,4 @@ After=swap.target
 What=tmpfs
 Where=/tmp
 Type=tmpfs
-Options=mode=1777,strictatime,nosuid,nodev,size=50%%,nr_inodes=1m
+Options=mode=1777,strictatime,nosuid,nodev,noexec,size=50%%,nr_inodes=1m
-- 
2.40.1

