From 060606c86791071663373a9ac45bd06cc3966552 Mon Sep 17 00:00:00 2001
From: Ben Cressey <bcressey@amazon.com>
Date: Sat, 28 May 2022 04:12:07 +0000
Subject: [PATCH] sysusers: set root shell to /sbin/nologin

Signed-off-by: Ben Cressey <bcressey@amazon.com>
Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>
---
 sysusers.d/basic.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sysusers.d/basic.conf.in b/sysusers.d/basic.conf.in
index 0aec080..d825aad 100644
--- a/sysusers.d/basic.conf.in
+++ b/sysusers.d/basic.conf.in
@@ -7,7 +7,7 @@
 
 # The superuser
 g root    0       -            -
-u root    0:0     "Super User" /root
+u root    0:0     "Super User" /root /usr/sbin/nologin
 
 # The nobody user/group for NFS file systems
 g {{NOBODY_GROUP_NAME}} 65534       -            -
-- 
2.40.1

