From c1fa329155c1b6805d64fcc7de4cb5c5ca80d387 Mon Sep 17 00:00:00 2001
From: Kush Upadhyay <kushupad@amazon.com>
Date: Thu, 13 Jun 2024 15:27:14 +0000
Subject: [PATCH] xioopts: conditionally compile applyopts_termios_value
 function

Signed-off-by: Kush Upadhyay <kushupad@amazon.com>
---
 xioopts.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xioopts.c b/xioopts.c
index 4b651aa..e957ba3 100644
--- a/xioopts.c
+++ b/xioopts.c
@@ -4041,6 +4041,7 @@ int applyopt_spec(
 	return 0;
 }
 
+#if WITH_TERMIOS
 int applyopts_termios_value(
 	int fd,
 	struct opt *opt)
@@ -4057,6 +4058,7 @@ int applyopts_termios_value(
 	 }
 	return 0;
 }
+#endif /* WITH_TERMIOS */
 
 /* Note: not all options can be applied this way (e.g. OFUNC_SPEC with PH_OPEN)
    implemented are: OFUNC_FCNTL, OFUNC_SOCKOPT (probably not all types),
-- 
2.40.1

