From a5f6b26082e0022d3c3e70e0718e4787939778d8 Mon Sep 17 00:00:00 2001
From: iliana destroyer of worlds <iweller@amazon.com>
Date: Tue, 30 Jul 2019 12:59:09 -0700
Subject: [PATCH] lustrefsx: Disable -Werror=stringop-overflow=

Signed-off-by: iliana destroyer of worlds <iweller@amazon.com>
---
 drivers/staging/lustrefsx/Makefile.rules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/lustrefsx/Makefile.rules b/drivers/staging/lustrefsx/Makefile.rules
index a0d56e80f2ce..62390580a3b4 100644
--- a/drivers/staging/lustrefsx/Makefile.rules
+++ b/drivers/staging/lustrefsx/Makefile.rules
@@ -3,4 +3,4 @@ ccflags-y += -include $(srctree)/drivers/staging/lustrefsx/config.h
 ccflags-y += -I$(srctree)/drivers/staging/lustrefsx/libcfs/include
 ccflags-y += -I$(srctree)/drivers/staging/lustrefsx/lnet/include
 ccflags-y += -I$(srctree)/drivers/staging/lustrefsx/lustre/include
-ccflags-y += -Wno-format-truncation -Werror
+ccflags-y += -Wno-format-truncation -Werror -Wno-error=stringop-overflow=
